fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/api/content/registry/v1/RarityProvider.java

```diff
@@ -14,14 +14,19 @@
  * limitations under the License.
  */
 
-package net.fabricmc.fabric.mixin.event.interaction;
+package net.fabricmc.fabric.api.content.registry.v1;
 
-import org.spongepowered.asm.mixin.Mixin;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.Formatting;
+import net.minecraft.util.Identifier;
 
-import net.minecraft.server.network.ServerPlayNetworkHandler;
+/**
+ * Allows giving an Item a custom rarity. This specifies the color of the Item's name when held or when hovered over.
+ * This will override the value returned in {@link Item#getRarity(ItemStack)}.
+ */
+public interface RarityProvider {
+	Identifier getId(ItemStack stack);
```

Comment:
why did this stay?

Reply:
not used anywhere internally. but it can be used.

Reply:
say I wanted to sort items based on their rarity. the identifier helps.

Reply:
This should've been removed.

Reply:
it's fine

Reply:
no, as you will be implementing this in your Item, without context what this even does and why this is a thing, and the name `getId` doesn't help

Reply:
how that this is merged we should probably move this to a new issue.

Reply:
fine

Reply:
fixed in #39

Reply:
no, I just give on this one.